File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int>t;
        for(string &s:tokens){
            if(s=="+"||s=="-"||s=="*"||s=="/"){
                long x=t.top();t.pop();
                long y=t.top();t.pop();
                if(s=="+")
                    t.push(x+y);
                else if(s=="-")
                    t.push(y-x);
                else if(s=="*")
                    t.push(x*y);
                else
                    t.push(y/x);
            }
            else{
                t.push(stoi(s));
            }
        }
        
        return t.top();
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        int len=tokens.length;
        
        Stack<Integer>st=new Stack<>();
        
        for(int i=0;i<len;++i){
            if(tokens[i].equals("+")){
                int y=st.pop();
                int x=st.pop();
                
                st.push(x+y);
            }
            else if(tokens[i].equals("-")){
                int y=st.pop();
                int x=st.pop();
                
                st.push(x-y);
            }
            else if(tokens[i].equals("*")){
                int y=st.pop();
                int x=st.pop();
                
                st.push(x*y);
            }
            else if(tokens[i].equals("/")){
                int y=st.pop();
                int x=st.pop();
                
                st.push(x/y);
            }
            else{
                
                st.push(Integer.valueOf(tokens[i]));
            }
        }
        
        return st.peek();
    }
}